exist-ant/src/main/java/org/exist/ant/XMLDBExtractTask.java

```diff
@@ -133,8 +134,8 @@ public void execute() throws BuildException {
      * @throws IOException if an I/O error occurs
      */
     private void extractResources(final Collection base, final String path) throws XMLDBException, IOException {
-        final String[] resources = base.listResources();
-        if (resources != null) {
+        final List<String> resources = base.listResources();
+        if (!resources.isEmpty()) {
```

Comment:
no risk for nPE?

Reply:
The API states that there no null results are allowed instead an empty list must be returned.